src/hooks/useDragLeave.ts

```diff
@@ -40,6 +41,16 @@ const useDragLeave = ({ isDeepHovering, canDropThought }: { isDeepHovering: bool
       return
     }
 
+    // If canDrop changed from true to false, immediately clear the hovering path
+    if (!canDropThought && prevCanDropRef.current) {
+      dispatch(clearHoveringPath)
+      prevCanDropRef.current = canDropThought
+      return
+    }
```

Comment:
This condition doesn't quite make sense to me. Isn't it strictly narrower than the condition that comes immediately after? i.e. `clearHoveringPath` will always be dispatched when `canDropThought` is false.

Reply:
Yep i can verify that the existing condition is redudant :)